staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/helpers.go

```diff
@@ -232,3 +232,14 @@ func HasObjectMetaSystemFieldValues(meta Object) bool {
 	return !meta.GetCreationTimestamp().Time.IsZero() ||
 		len(meta.GetUID()) != 0
 }
+
+// ResetObjectMetaForStatus forces the meta fields for a status update to match the meta fields
+// for a pre-existing object. Status updates are for updating status, not metadata.
+func ResetObjectMetaForStatus(meta, existingMeta Object) {
```

Comment:
here or in the generic registry?

Reply:
> here or in the generic registry?

Looks more registry-ish

Reply:
I agree this should go near where it is used. It is always easy to promote it to a more visible location if a second user comes along, but it's very hard to go the other direction.

Reply:
Done.